config/static.c

```diff
@@ -26,103 +35,121 @@
 #include "gatekeeper_lls.h"
 #include "gatekeeper_rt.h"
 
+#define RTE_LOGTYPE_CONFIG RTE_LOGTYPE_USER2
+
+/* TODO Get the install-path via Makefile. */
+#define LUA_BASE_DIR "./lua/"
+#define LUA_CONFIG_FILE     "gatekeeper_config.lua"
+
+struct gatekeeper_config {
+	struct bp_config 	*bp_conf;
+	struct catcher_config 	*catcher_conf;
+	struct dynamic_config 	*dy_conf;
+	struct cps_config 	*cps_conf;
+	struct ggu_config 	*ggu_conf;
+	struct gk_config 	*gk_conf;
+	struct gt_config 	*gt_conf;
+	struct rt_config 	*rt_conf;
+};
+
+static lua_State *lua_state;
```

Comment:
Each instance of a block has a dedicated lcore, so when an instance sees the `exiting` variable true, it will destroy its resources and exit. It's my understanding that the function `rte_eal_mp_wait_lcore()` in main/main.c only returns only when all lcores are released.

With this information in mind, why does one need anything else?